plugins/package-managers/node/src/main/kotlin/Npm.kt

```diff
@@ -22,68 +22,41 @@
 package org.ossreviewtoolkit.plugins.packagemanagers.node
 
 import java.io.File
-import java.util.concurrent.ConcurrentHashMap
```

Comment:
Commit message:

> While the same is planned for `Yarn` too, this change is just a refactoring step which allows to do `Npm` first.

As this refactoring moves around so much code which is hard to review, it would have been nice to explain *why* `Npm` is done first. Wouldn't it have been easier to do `Yarn` first?

Reply:
Doing this for `Npm` is way more straight forward than doing this for `Yarn`. 
I wanted to do the low hanging fruit first, which also allows to quickly add workspaces support for NPM. 

How to best implement this in Yarn I'm not sure. I'm considering to ask Yarn community whether they are open to a contribution to the `list` command, so that it outputs something which is less complex to process for our purpose.

Reply:
> I'm considering to ask Yarn community whether they are open to a contribution to the `list` command

But then we would depend on (at least) that Yarn version, which is probably not feasible anyway.

Reply:
Hm, why would this be problematic. I do remember similar thing for `go mod`.

Reply:
Because projects might depend on a lower version of Yarn (for whatever reason), similar to like some projects depend on a lower Version of Gradle.